### PR TITLE
Discovery UI

### PR DESCRIPTION
**Pull request by rgbkrk**
This is paving a path to show off published notebooks.

Only a prototype, certainly not in the state we'll want it yet. Figured it would be good for me to post publicly where things are at. Have done _zero_ styling of the components here, this is default semantic UI. Mostly just playing with the semantic UI components, seeing what the structure and design is.

screen shot 2017-02-20 at 10 02 53 pm

xref: https://github.com/nteract/nteract/issues/1465

### CODE REVIEW

**Comment by cabhishek** [comment]
@rgbkrk looks like build is failing due to [ES lint errors](https://travis-ci.org/nteract/commuter/builds/203701984#L209)

**Comment by rgbkrk** [comment]
Oh are you wanting to merge this as I have it now? This is all junk data at the moment, happy to run prettier.

**Comment by cabhishek** [comment]
@rgbkrk I wasn't sure if you wanted to merge or keep this open while you work on it.

**Comment by rgbkrk** [comment]
Yeah it's still a Work in Progress (WIP).

**Comment by rgbkrk** [comment]
Leaving as a note for myself for later, I like the responsiveness of this design: http://codepen.io/sdras/full/7f31b37ae59639b3f7f66b76bd6f010d

**Comment by rgbkrk** [comment]
Strawman for a top level Discovery API:

```js
GET /api/v1/discover

{
  "results": [
    {
      "publishedOn": "February 20, 2016",
      "image": "https://cloud.githubusercontent.com/assets/836375/23152279/e945b5ee-f7b5-11e6-9798-9bffa5026aa7.png",
      "path": "/example-notebooks/display-updates.ipynb",
      "metadata": {
        "title": "Named Display Updates",
        "authors": [{"name": "Paul Ivanov"}, {"name": "Kyle Kelley"}],
        "nteract": {
          "tags": ["ipython", "display updates", "progress-bars"],
          "description": "A Progressive look at updating outputs programmatically from IPython"
        },
        "kernelspec": {
          "name": "python3",
          "language": "python",
          "display_name": "Python 3"
        },
        "kernel_info": {
          "name": "python3"
        },
        "language_info": {
          "codemirror_mode": {
            "name": "ipython",
            "version": 3
          },
          "nbconvert_exporter": "python",
          "mimetype": "text/x-python",
          "version": "3.5.2",
          "file_extension": ".py",
          "name": "python",
          "pygments_lexer": "ipython3"
        }
      }
    }
  ]
}
```

**Comment by cabhishek** [comment]
this [PR](https://github.com/nteract/commuter/pull/70) has the necessary mapping details (creation and schema)

**Comment by rgbkrk** [comment]
Rebasing this again, I'll incorporate what I did last night with the exposed API + elasticsearch.

File: packages/commuter-client/src/discovery.json [hunk 1]
```diff
@@ -0,0 +1,101 @@
+[
```
Comment (cabhishek) [inline packages/commuter-client/src/discovery.json]: is this file user anywhere?

Comment (rgbkrk) [inline packages/commuter-client/src/discovery.json]: Not anymore, let me delete this and rebase+squash.

**Comment by cabhishek** [comment]
Looks good. Ready for merge?

**Comment by rgbkrk** [comment]
Yeah this is ready for merge. I'm doing some more UI tweaks to match the original mockup, they can come in a new PR though. Thanks!

**Pull request merged**